### DISCUSSION TITLE
Prevent graphql errors if no content is published in Contentful

### DISCUSSION THREAD

**Discussion by bagseye** [gatsbyjs/gatsby#32009] [Help] [answered]
I created the following theme [Barcadia](https://barcadia.netlify.app/) [(Repo)](https://github.com/bagseye/barcadia) about 12 months ago, but had a message from someone looking to use it saying they had multiple errors when running `gatsby-develop`. 

It turned out the errors are related to no content being present in Contentful so when it runs the build process, it cannot build the page templates set out in `gatsby-node.js`. The terminal suggests creating 'dummy content' to counter this, but from what I understand, I need to declare types to prevent errors if no 'dummy content' is present.

I'm quite unfamiliar with graphql and my knowledge is minimal. I'm not entirely clear on what is required to achieve this using the [Gatsby docs](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#creating-type-definitions)  But this is what I have so far: 

Query found in `blog-list-template.js`
```
query getPosts($skip: Int!, $limit: Int!) {
    post: allContentfulPosts(
      skip: $skip
      limit: $limit
      sort: { fields: published, order: DESC }
    ) {
      edges {
        node {
          slug
          title
          introduction
          postId: contentful_id
          published(formatString: "Do MMMM YYYY")
        }
      }
    }
  }
`
```
Types set in `gatsby-node.js`
```
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions

  const typeDefs = `
  type ContentfulPosts implements Node {
    title: String
    slug: String
    introduction: String
    published(formatString: Date = "Do MMMM YYYY"): String 
    richText: BlogContent
  }
  type BlogContent {
    raw: [String]
  }

  `

  createTypes(typeDefs)
}
```

Am I on the right track with this? I appear to be making some progress with types that are not nested, but once dealing with nesting and arguments there seems to be real issues.

Also, is this the best way to do this? Basically I am trying to create a theme that will not display errors if the user has not populated the content.

Any info or advice would be much appreciated.

Morgan

**Comment by axe312ger** [comment, marked as answer]
Yes, you probably need to define your types manually. Also you could use https://www.gatsbyjs.com/plugins/gatsby-plugin-schema-snapshot/ which generates these types for you. 

This will be fixed with the next big major version. You can follow the development process here: #31385
